doc/dependency-resolution.md

```diff
@@ -1,356 +1,129 @@
-need only place this content item on a 'dirty' queue for processing by the
-dependency resolution worker.
+At the point of representing a content item to a Content Store we need to load
+the details of every content item that has an explicit or implicit link from
+this content item.
 
-If required, we can separate the calculation of dependencies from sending the
-updates downstream into separate workers/queues. The important thing is that
-both of those tasks happen out-of-band to the request cycle.
+Explicit links are links which have been set for a content item through a
+PatchLinkSet request. eg a request of:
 
-Note: Bidirectional dependencies is documented separately later on.
+@TODO example
 
-- Link expansion
+Would store a link to 123 of type:
 
-It is proposed that we move link expansion into the publishing api and send the
-fully-pressed content item downstream to the content store. For each of the link
-groups that appear on the content item, it will need to look up the rules for
-those groups from the schemas and expand those fields from the dependent content
-items.
+@TODO exmaple
 
-We also need to consider a new data structure to represent recursively expanded
-links. It is proposed that the structure remain almost identical with the key
-difference that the links for dependent content items are recursively expanded
-in the links data. For now, I think we should include the expanded links as a
-separate field in the content item. Perhaps this can be called "expanded_links",
-or similar and we can send it alongside the unexpanded links. As we migrate
-applications to use dependency resolution, we should make them use this field
-instead of "links" and look at retiring the expansion of links in content store.
+Thus any time either 456 or 123 changes a new representation of the content
+item is created for the Content Store.
 
-Here is a rough example of what this **might** look like:
+In addition to explicit links there are also implicit links between content
+items. These occur in 3 scenarios:
 
-```javascript
-{
-  "expanded_links": {
-    "parent": [
-      {
-        "content_id": "111-111",
-        "title": "VAT Rates",
-        "base_path": "/tax/vat-rates",
-        "parent": [
-          {
-            "content_id": "222-222",
-            "title": "Taxation",
-            "base_path": "/tax",
-            "parent": [
-              {
-                "content_id": "333-333",
-                "title": "Home",
-                "base_path": "/"
-              }
-            ]
-          }
-        ]
-      }
-    ]
-  }
-}
-```
+- As translations. If a content item is available with a different locale it
+  will automatically be linked as a translation of a content item.
+- Content items linking to this item as a parent item. Content items that are
+  linked to as a parent automatically implicitly reciprocate this linking as
+  children of this link.
```

Comment:
The implied "children" link type is only one of the 4 implicit link types, it might be good to try and generalise the description, and just use parent/children as an example.